reana_client/utils.py

```diff
@@ -59,23 +60,28 @@ def load_workflow_spec(workflow_type, workflow_file, **kwargs):
     return workflow_load[workflow_type](workflow_file, **kwargs)
 
 
-def load_reana_spec():
+def load_reana_spec(filepath, skip_validation=False):
```

Comment:
Regarding specification validation I would rather make `_validate_reana_yaml` a public function and make another command `reana validate -f <spec_file>` which uses it.

Inside `load_reana_spec` I would make validation mandatory.

Reply:
> make another command reana validate -f <spec_file> which uses it.

Yes. It's coming in https://github.com/hjhsalo/reana-client/blob/19_add_validate_command/reana_client/cli/analyses.py
No PR opened yet, since I want to see what gets merged to master by this PR.

> _validate_reana_yaml a public function
> Inside load_reana_spec I would make validation mandatory.

If I make changes to `_validate_reana_yaml`, I need to reuse the yaml loading part from `load_reana_spec` either in calling function or in the new `validate_reana_yaml`?

Reply:
If we move `yaml.load()` inside `validate_reana_yaml` I would say that this function returns the spec as a dict so you can use it in `load_reana_spec`. But if we do so, `validate_reana_yaml` and `load_reana_spec` are the same function in the end so there is no need to have two, so we have two options:

1. We go for just one function which includes both loading and validation specific code, so when we want to validate or load we call the same.
2. We keep then as they are, and we always call `load_reana_yaml` either for validating or loading but having place for validation code and another for loading.

The difference for me lies in future changes to validation and loading keeping concerns separate or in the same function. I would go for the second 😄